protocol/src/fixed_codec/epoch.rs

```diff
@@ -0,0 +1,231 @@
+use bytes::Bytes;
+
+use crate::{ProtocolResult, impl_default_fixed_codec_for};
+use crate::fixed_codec::{FixedCodecError, ProtocolFixedCodec};
+use crate::types::{
+    Bloom,
+    primitive::{Hash, UserAddress, MerkleRoot},
+    epoch::{Proof, Validator, Epoch, EpochHeader, Pill},
+};
+
+impl_default_fixed_codec_for!(epoch, [Proof, Validator, Epoch, EpochHeader, Pill]);
+
+impl rlp::Encodable for Proof {
+    fn rlp_append(&self, s: &mut rlp::RlpStream) {
+        s.begin_list(5)
+            .append(&self.bitmap.to_vec())
+            .append(&self.epoch_hash)
+            .append(&self.epoch_id)
+            .append(&self.round)
+            .append(&self.signature.to_vec());
+    }
+}
+
+impl rlp::Decodable for Proof {
+    fn decode(r: &rlp::Rlp) -> Result<Self, rlp::DecoderError> {
+        if !r.is_list() && r.size() != 5 {
+            return Err(rlp::DecoderError::RlpIncorrectListLen);
+        }
+
+        let mut values = Vec::with_capacity(5);
+
+        for val in r {
+            let data = val.data()?;
+            values.push(data)
+        }
+
+        let bitmap = Bytes::from(values[0]);
+        // let epoch_hash = Hash::from_bytes(Bytes::from(values[1]))
+        //    .map_err(|_| rlp::DecoderError::RlpInvalidLength)?;
+        let epoch_hash: Hash = rlp::decode(r.at(1)?.as_raw())?;
+        let epoch_id = r.at(2)?.as_val()?;
+        let round = r.at(3)?.as_val()?;
+        let signature = Bytes::from(values[4]);
+
+        Ok(Proof {
+            epoch_id,
+            round,
+            epoch_hash,
+            signature,
+            bitmap
+        })
+    }
+}
```

Comment:
The implement of `rlp::Decodable` may refer to this https://github.com/cryptape/overlord/blob/master/src/codec.rs#L19:L34